optuna_dashboard/_note.py

```diff
@@ -110,6 +110,23 @@ def note_str_key_prefix(trial_id: Optional[int]) -> str:
     return f"dashboard:{trial_id}:note_str:"
 
 
+def transfer_notes(storage: BaseStorage, src_study: optuna.Study, dst_study: optuna.Study) -> None:
+    system_attrs = storage.get_study_system_attrs(study_id=src_study._study_id)
+
+    def transfer(src_trial_id: Optional[int], dst_trial_id: Optional[int]) -> None:
+        note = get_note_from_system_attrs(system_attrs, src_trial_id)["body"]
+        save_note_with_version(storage, dst_study._study_id, dst_trial_id, 0, note)
+        delete_notes(storage, src_study._study_id, src_trial_id)
+
+    # Transfer individual trial notes
+    for src_trial, dst_trial in zip(src_study.get_trials(), dst_study.get_trials()):
+        transfer(src_trial._trial_id, dst_trial._trial_id)
+
+    # Transfer study note
+    NO_SRC_TRIAL, NO_DST_TRIAL = None, None
+    transfer(NO_SRC_TRIAL, NO_DST_TRIAL)
```

Comment:
I would say we don't need to delete original notes. How about making it `copy` instead of `transfer` since we can reduce the number of storage API calls?

```suggestion
def copy_notes(storage: BaseStorage, src_study: optuna.Study, dst_study: optuna.Study) -> None:
    system_attrs = storage.get_study_system_attrs(study_id=src_study._study_id)

    # Copy individual trial notes
    for src_trial, dst_trial in zip(src_study.get_trials(), dst_study.get_trials()):
        note = get_note_from_system_attrs(system_attrs, src_trial._trial_id)["body"]
        save_note_with_version(storage, dst_study._study_id, dst_trial._trial_id, 0, note)

    # Copy study note
    note = get_note_from_system_attrs(system_attrs, None)["body"]
    save_note_with_version(storage, dst_study._study_id, None, 0, note)
```